### PR TITLE
[TS] LPS-149303 Recurring All Day Calendar Event are skipping months when the Time Zone is not UTC

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Thank you and best regards,
István

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Hi @IstvanD,

I've tested and you fix the bug :tada:

now I will review the code to try to find something that might be broken

Thank you so much for your help,
Feliphe

File: modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ private Calendar _getStartTimeJCalendar(DateValue dateValue) {
 			_getTimeZone(_calendarBooking));
 
 		int shift = JCalendarUtil.getDSTShift(
-			jCalendar, startTimeJCalendar, _displayTimeZone);
+			jCalendar, startTimeJCalendar, _getTimeZone(_calendarBooking));
```
Comment (MarinhoFeliphe) [inline modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:110]: Hi @IstvanD,

could you replace `_getTimeZone`... for this:

`(_calendarBooking.isAllDay()) ? TimeZone.getTimeZone(StringPool.UTC) : _displayTimeZone`

Thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
Resent here #290

File: modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:110]: hi @MarinhoFeliphe thanks again for checking! I see that you replaced the method for the expression in your new pull request. I assume I don't have to do it at this point. Is that correct?

Comment (MarinhoFeliphe) [reply modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java:110]: Exactly, @IstvanD

I'm just trying to find the cause of the other errors, I'll keep you updated.

Thanks!

**Pull request closed without merging**